overlays/cross-compilation.nix

```diff
@@ -5,6 +5,41 @@
 #
 {...}: {
   nixpkgs.overlays = [
-    (final: prev: {})
+    (final: prev: let
+      crossCompiling = prev.stdenv.buildPlatform != prev.stdenv.hostPlatform;
+      filterOutByName = name: builtins.filter (x: (builtins.baseNameOf x) != name);
+      # FIXME: should be prev.buildPackages.fetchpatch2, why I can't use fetchpatch2 here
```

Comment:
Does final.buildPackages.fetchpatch2 work? Or final.fetchpatch2?

Reply:
doesn't worked for me, but I believe GH yield stable commits, so we can hash it

Reply:
I don't know if I will soon hit some other problem, but using both final.buildPackages.fetchpatch2 and prev.buildPackages.fetchpatch2 at least evaluates for me, and starts building. Maybe there's error to come back to